server/belga/search_providers.py

```diff
@@ -375,6 +376,26 @@ def find(self, query, params=None):
 
         data = self.api_get(self.search_endpoint, api_params)
         docs = [self.format_list_item(item) for item in data[self.items_field]]
+
+        # SDBELGA-667
+        if search_text := api_params.get("searchText"):
+            search_text = "|".join(search_text.split())
+            fields = ("body_html", "headline", "slugline")
+            for doc in docs:
+                for field in fields:
+                    highlighted_value = re.subn(
+                        f"({search_text})",
+                        lambda text: " ".join(
+                            [
+                                f'<span class="es-highlight">{s}</span>'
+                                for s in text.groups()
+                            ]
+                        ),
+                        doc.get(field),
+                    )
```

Comment:
would probably also use `flags=re.IGNORECASE` so it will match both cases, looks like the search is not case sensitive